### ISSUE TITLE
Refining fiscal year transformations on domestic data

### ISSUE THREAD

**Issue by Duncan-Knox** [devinit/datahub#444]
Aim to use transformation data aligned from calendar year to fiscal year to use for countries which have government finance data in fiscal year which differ to the calendar year. This would improve the quality of data through mitigating impact of inflation between the months not aligned for fiscal and calendar years.

This applies to:

Bangladesh, Bhutan, Dominica, Egypt, Ethiopia, Kenya, Malawi, Mauritius, Nepal, Pakistan, Rwanda, Samoa, South Sudan, Tanzania, Tonga, Uganda (July to June). Burundi has just changed to July-June FY also..

**Comment by danwdevinit** [comment]
Datasets used (all from IMF WEO):
- GDP Fiscal Year (NCU current prices)
- GDP Calendar Year (USD current prices)
- GDP Calendar Year (Percent growth)

1) Calculate **GDP CY (USD constant 2016)** from **GDP CY (USD current)** and **GDP CY growth** 
2) Conversion factor is then [**GDP CY (USD constant 2016)**] / [**GDP FY (NCU current)**]

**Comment by danwdevinit** [comment]
> **6 (Bhutan, Egypt, Ethiopia, Nepal, Pakistan and Samoa)** only have data for FY in the WEO, so presume the defaltor data is also in FY

Yes, the deflator is FY. We therefore need to calculate an appropriate CY base 2016 deflator.

> **4 (Bangladesh, Dominica, Malawi and Mauritius)** have NCU current GDP in FY, so we would need to collect US$ GDP also. Guessing deflators will therefore be in CY.

For these 4 countries, the above conversion method will work.

> **6 (Kenya, Rwanda, South Sudan, Tanzania, Tonga and Uganda)** only have data for CY in the WEO, so we would need to collect GDP in NCU and US$. Guessing deflators will therefore be in CY.

These 6 will prove more tricky - without access to any raw FY data from IMF, we will need to look at other sources.

**Comment by elmetodo** [comment]
Hi Dan,

Apologies I am slightly confused, maybe best to have a quick call. As the government finance data (NCU current) is in FY the idea was to have the transformation data be in FY too, as assuming was it was all CY. However, for the first 6 there is basically no need to change anything as the transformation data is in FY already.

On the last 6 this isn't too much of an issue, as there is NCU FY GDP (current) data in the IMF staff/program reports. Likely US$ GDP FY (current) in them also and GDP FY growth.

**Comment by danwdevinit** [comment]
> As the government finance data (NCU current) is in FY the idea was to have the transformation data be in FY too, as assuming was it was all CY. However, for the first 6 there is basically no need to change anything as the transformation data is in FY already.

I may be the one who's mistaken... I believe we are looking to have the data (FY or otherwise) in constant 2016 USD prices, which would mean we need to be able to deflate to CY 2016.

> On the last 6 this isn't too much of an issue, as there is NCU FY GDP (current) data in the IMF staff/program reports. Likely US$ GDP FY (current) in them also and GDP FY growth.

Ah yes. For some reason I assumed the WEO would contain these FY data. You're definitely right that we can extract them from the programme reports.

Definitely happy to have a call - but I am quite croaky with my current cold.

**Comment by Duncan-Knox** [comment]
This is an ongoing task via @elmetodo and @danwdevinit and unlikely to impact developer capacity for the time being.

**Comment by Duncan-Knox** [comment]
Postponed into the new year. Added to agenda on discussion with Richard Watts on 11/12/2018.
